caikit_nlp/toolkit/text_generation/model_run_utils.py

```diff
@@ -233,10 +233,19 @@ def generate_text_func(
         for g in generate_ids
     ]
 
-    if generate_ids[0][-1].item() == eos_token:
+    if (eos_token and tokenizer.decode(generate_ids[0, -1].item()) == eos_token) or (
+        generate_ids[0, -1] == tokenizer.eos_token_id
+    ):
         finish_reason = "EOS_TOKEN"
     elif generate_ids.size(1) - 1 == max_new_tokens:
         finish_reason = "MAX_TOKENS"
+    elif ("stopping_criteria" in gen_optional_params) and (
+        gen_optional_params["stopping_criteria"](
+            generate_ids,
+            None,  # scores, unused by SequenceStoppingCriteria
+        )
+    ):
+        finish_reason = "STOP_SEQUENCE"
     else:
```

Comment:
Lets remove `OTHER` since it creates issues anyways?

Reply:
What do you propose to do in that case? I'd rather have some way of handling the else branch (`RuntimeError`? )

Reply:
Yeah, we can do `error("<>", ...)` that way its evident something went wrong

Reply:
What error code should be using though? I think the exception should be something in the lines of `RuntimeError("Could not determine finish reason")`

Reply:
Yep. An example of throwing this via `error` : 

https://github.com/caikit/caikit/blob/main/caikit/core/data_model/streams/data_stream.py#L247


The error code over there, i.e `<COR59442457E>` is of the format: `<3 char lib identifier><8 int random number>E`

1. `<3 char lib identifier>` for caikit nlp is `NLP`
2. E suffix identifies this as a ERROR level message

Reply:
@gkumbhat I was wondering where those error numbers come from.  So something like this to generate on mac is okay?

```ShellSession
markstur@Marks-MacBook-Pro-3 ~ % echo $RANDOM$RANDOM
439318353
```

And whatever padding/truncation you like to grab 8 random-looking digits?  Wierd... but that should help search code.

Reply:
@gkumbhat Thanks. Updated the code with `error("<NLP38342991E>", RuntimeError(...))`

@markstur 
In python: 
```python
from random import choices

lib_identifier = "NLP"
message_type = "E"
unique_code = "".join(map(str, choices(range(10), k=8)))

print(f"{lib_identifier}{unique_code}{message_type}")
```

Reply:
If you use vscode, you can use this extension to automatically fill appropriate log code: https://marketplace.visualstudio.com/items?itemName=Gaurav-Kumbhat.alog-code-generator